libevmasm/Assembly.cpp

```diff
@@ -242,13 +246,101 @@ string Assembly::toStringInHex(u256 _value)
 	hexStr << std::uppercase << hex << _value;
 	return hexStr.str();
 }
+vector<Json::Value> Assembly::assemblyItemAsJSON(AssemblyItem const& _item) const
```

Comment:
Function names here are confusing to me. I had to read `createJsonValue()` and `assemblyItemAsJSON()` to really understand what they do. From the names and arguments alone, they both look like they're just translating an `AssemblyItem` to JSON.

It would be really helpful to have a term for the thing that `createJsonValue()` returns. Can we call it an opcode? Assembly instruction? Line of assembly? It's a smaller unit than assembly item but it also includes tags which aren't opcodes/instructions. We should use that term in the name.

Reply:
Yeah you where totally right. I removed `createJsonValue()` now. I think after I did https://github.com/ethereum/solidity/pull/12799#issuecomment-1070640194 the code is now more readable.